sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuShuffleEnv.scala

```diff
@@ -104,11 +104,24 @@ object GpuShuffleEnv extends Logging {
   // this forces the initialization when we know we are ready in the driver and executor.
   //
   def initShuffleManager(): Unit = {
-    SparkEnv.get.shuffleManager match {
+    val shuffleManager = SparkEnv.get.shuffleManager
+    shuffleManager match {
       case rapidsShuffleManager: RapidsShuffleManagerLike =>
         rapidsShuffleManager.initialize
       case _ =>
-        throw new IllegalStateException(s"Cannot initialize the RAPIDS Shuffle Manager")
+        val caseClassLoader =
+          classOf[RapidsShuffleManagerLike].getClassLoader
+        throw new IllegalStateException(s"Cannot initialize the RAPIDS Shuffle Manager: " +
+        s"RapidsShuffleManagerLike class used class loader: " +
+        s"${caseClassLoader.toString}, with hash code: ${caseClassLoader.hashCode}. " +
+        s"ShuffleManager instance contains the following RapidsShuffleManagerLike class loader: " +
+        s"${shuffleManager.getClass.getSuperclass.getInterfaces
+          .collect {
+            case c if c.getName == classOf[RapidsShuffleManagerLike].getName =>
+              s"${c.getClassLoader} with hash code: ${c.getClassLoader.hashCode}"
+          }.mkString("Array(", ", ", ")")
+        }"
+        )
```

Comment:
The expression on L118 is too complex to simply interpolate it into an s-string.

I think the error message should read something like this:

```suggestion
        val rsmLoaderViaShuffleManager = shuffleManager.getClass.getSuperclass.getInterfaces.collectFirst {
          case c if c.getName == classOf[RapidsShuffleManagerLike].getName => c.getClassLoader
        }
        val rsmLoaderDirect = classOf[RapidsShuffleManagerLike].getClassLoader
 
        throw new IllegalStateException(s"Cannot initialize the RAPIDS Shuffle Manager ${shuffleManager}! Expected: an instance of  RapidsShuffleManagerLike load by ${rsmLoaderDirect}. Actual: ${shuffleManager}  tagged with RapidsShuffleManagerLike loaded by: ${rsmLoaderViaShuffleManager}"
        )
```